pkg/sql/sem/tree/cast.go

```diff
@@ -1001,6 +1369,14 @@ var castMap = map[oid.Oid]map[oid.Oid]cast{
 				"using to_char(t AT TIME ZONE 'UTC') instead.",
 		},
 	},
+	oid.T__timestamptz: {
+		// Automatic I/O conversions to string types.
+		oid.T_bpchar:  {maxContext: CastContextAssignment, origin: contextOriginAutomaticIOConversion, volatility: VolatilityStable},
```

Comment:
i wonder if it's worth making utility that adds all the strings ones at once, e.g.

```
applyDefaultForStrings(
   map[oid.Oid]cast{...},
  ({maxContext: CastContextAssignment, origin: contextOriginAutomaticIOConversion, volatility: VolatilityStable})...
)

func applyDefaultForStrings(o map[oid.Oid]cast, c cast) map[oid.Oid]cast {
   for _, stringOID := range []oid.Oid{oid.T_bpchar, ...} {
       o[stringOID] = c
    }
   return o
}
```

up to you though.

Reply:
I think I'll leave as-is for this PR, but definitely something we can refine later.